### PR TITLE
Added multiple new features!

### PR DESCRIPTION
**Pull request by Internal3247**
Added multiple new features and did some bug fixes/formatting:

- Idle light/park lights
- Blackout mode (disables brake lights, headlights, idle lights if the vehicle has them, emergency lights, sirens, interior lights, plate lights, indicators) Automatic siren cutoff (siren turns off automatically when exiting vehicle) 
- Siren tones can only be toggled on stage 3 for bozlight-enabled vehicles) 
- Fixed a bug where bozlight-enabled vehicles are constantly running their siren when out of onesync distance

### CODE REVIEW

**Review by OfficerBozza — COMMENTED** [review]
I think Siren Park Kill and Blackout Mode should just apply to any vehicles that are either in the emergency class, or the ModelsWithEquipment list. It would save the end user from having to list out basically every vehicle, because these are common enough features that they could be applied to every car. 

Otherwise these lists could be inverted, so cars that meet the above have it by default, but you can list vehicles that *shouldn't* have it if you want any exceptions

**Comment by Internal3247** [comment]
The inverted idea is good. I think ill do that. I did it originally since older vehicles don't have those features. Making a new PR soon with those features along with messageboard and TA functionality so ill include it then. 👍

File: client.lua [hunk 1]
```diff
@@ -1353,7 +1415,43 @@ Citizen.CreateThread(function()
 						end
 					end
 
-					
+					-- siren park kill
+					if UseParkKill(veh) then
+						if IsControlPressed(0, 75) then
+							SetLxSirenStateForVeh(veh, newstate)
```
Comment (szajkok) [inline client.lua:1421]: Here's an alternative if you'll have issues with siren sync:
```lua
Citizen.CreateThread(function()
	while true do
		Citizen.Wait(0)
		local ped = GetPlayerPed(-1)
		local veh = GetVehiclePedIsIn(ped, true)
		--if  player_is_emerg_driver then
			--if  IsPedSittingInVehicle(ped, veh) and GetPedInVehicleSeat(veh, -1) == PlayerPedId() and ParkKill(veh) and GetIsTaskActive(ped, 2) and GetVehiclePedIsIn(ped, true) then
			if  GetPedInVehicleSeat(veh, -1) == ped and IsPedInAnyVehicle(ped, false) and ParkKill(veh) and GetIsTaskActive(ped, 2) and state_lxsiren[veh] ~= 0 then
				print("po")
				TriggerServerEvent('InteractSound_SV:PlayOnSource', 'shifterleo', 0.3)
				SetLxSirenStateForVeh(veh, 0)
				SetAirManuStateForVeh(veh, 0)
				
				count_bcast_timer = delay_bcast_timer
				Citizen.Wait(1000)
			--end
		end
	end
end)
```

I assume that it's a wacky code but it does the job, if someone would like to clean it up - you're more than welcome to do so.

**Comment by szajkok** [comment]
alr so I came across another issue with the script, after you add `and lstate == 3 ` to the `SetLxSirenStateForVeh` function, sirens on non-stage cars dont work

**Comment by szajkok** [comment]
soo idk if that's how I should make it work but if you just edit this line of code:
```lua
								-- BOZLIGHTS END
									if IsVehicleSirenOn(veh) then
										PlaySoundFrontend(-1, "NAV_UP_DOWN", "HUD_FRONTEND_DEFAULT_SOUNDSET", 1)
										SetVehicleSiren(veh, false)
									else
										PlaySoundFrontend(-1, "NAV_LEFT_RIGHT", "HUD_FRONTEND_DEFAULT_SOUNDSET", 1)
										SetVehicleSiren(veh, true)
										count_bcast_timer = delay_bcast_timer
									end
								end	
```
into this:
```lua
								-- BOZLIGHTS END
									if IsVehicleSirenOn(veh) then
										PlaySoundFrontend(-1, "NAV_UP_DOWN", "HUD_FRONTEND_DEFAULT_SOUNDSET", 1)
										SetVehicleSiren(veh, false)
										lstate = 0
									else
										PlaySoundFrontend(-1, "NAV_LEFT_RIGHT", "HUD_FRONTEND_DEFAULT_SOUNDSET", 1)
										SetVehicleSiren(veh, true)
										lstate = 3
										count_bcast_timer = delay_bcast_timer
									end
```									
by just adding `lstate = 3` and `lstate = 0` it seems to work

**Comment by Internal3247** [comment]
Noted and being fixed in the next version of this, expect an update within the next week or two.

**Comment by JackReagan** [comment]
Any Updates ? Because if I follow szajkok instructions I lost almost all my FPS with this chunk of code: 

Citizen.CreateThread(function()
	while true do
		Citizen.Wait(0)
		local ped = GetPlayerPed(-1)
		local veh = GetVehiclePedIsIn(ped, true)
		--if  player_is_emerg_driver then
			--if  IsPedSittingInVehicle(ped, veh) and GetPedInVehicleSeat(veh, -1) == PlayerPedId() and ParkKill(veh) and GetIsTaskActive(ped, 2) and GetVehiclePedIsIn(ped, true) then
			if  GetPedInVehicleSeat(veh, -1) == ped and IsPedInAnyVehicle(ped, false) and UseParkKill(veh) and GetIsTaskActive(ped, 2) and state_lxsiren[veh] ~= 0 then
				print("po")
				TriggerServerEvent('InteractSound_SV:PlayOnSource', 'shifterleo', 0.3)
				SetLxSirenStateForVeh(veh, 0)
				SetAirManuStateForVeh(veh, 0)
				
				count_bcast_timer = delay_bcast_timer
				Citizen.Wait(1000)
			--end
		end
	end
end)

And if I'm not using it there is a real problem with the sync of siren, like responding and arriving from far and no one is hearing the siren or the siren is just the vanilla one.

**Comment by Internal3247** [comment]
Open client.lua of bozlights, go down to line 1421 and replace everything there with this:

                    -- siren park kill
                    if UseParkKill(veh) then
                        if IsControlPressed(0, 75) then
                            SetLxSirenStateForVeh(veh, newstate)
                            SetLxSirenStateForVeh(veh, 0)
                            state_lxsiren[veh] = 0
                            count_bcast_timer = delay_bcast_timer
                        end
                    end

**Comment by Internal3247** [comment]
Im working on a new version with multiple new features and bug fixes but its taking a bit longer then expected. The code I just posted works perfectly fine for myself and other players on my own server, so it should work for you with no issue. Let me know if that fixes your problem.

**Comment by szajkok** [comment]
I'm going to test it out again tommorow and let you know.

**Comment by Internal3247** [comment]
Another issue that could be causing this is the "if lstate == 3 then" next to the part where the sirens are turned on. I did this to prevent the ambient siren bug but it also seems to have issues with syncing player to player.

**Comment by JackReagan** [comment]
> Open client.lua of bozlights, go down to line 1421 and replace everything there with this:
> 
> ```
>                 -- siren park kill
>                 if UseParkKill(veh) then
>                     if IsControlPressed(0, 75) then
>                         SetLxSirenStateForVeh(veh, newstate)
>                         SetLxSirenStateForVeh(veh, 0)
>                         state_lxsiren[veh] = 0
>                         count_bcast_timer = delay_bcast_timer
>                     end
>                 end
> ```

Yep I've tried it for a couple of weeks and It's really better but recently all the almost all the siren are not sync/not playing, the only thing synced correctly is the horn. Btw I've no FPS drops so yeah better ! Idk where is the "if lstate == 3 then" you are referring if you can tell me ? And thanks for the help ^^

**Comment by JackReagan** [comment]
Also to disable the distant siren just add down CleanupSounds() this "DistantCopCarSirens(false)" and the ambiant siren are disable now ^^

Should look something like this 

```
Citizen.CreateThread(function()
	while true do
			
			CleanupSounds()
			
			DistantCopCarSirens(false)

			RequestScriptAudioBank("DLC_POLICINGMPAUDIO\\POLICINGMP_SIRENS1", false)
			RequestScriptAudioBank("DLC_POLICINGMPAUDIO\\POLICINGMP_SIRENS2", false)
			RequestScriptAudioBank("DLC_POLICINGMPAUDIO\\POLICINGMP_SIRENS3", false)
			RequestScriptAudioBank("DLC_POLICINGMPAUDIO\\POLICINGMP_SIRENS4", false)
			RequestScriptAudioBank("DLC_POLICINGMPAUDIO\\POLICINGMP_SIRENS5", false)
```

**Comment by Internal3247** [comment]
@JackReagan, thanks for your help with the ambient siren sounds. I've tested it on my own server and it seems to work flawlessly. With the "if lstate == 3 then", go to lines 763, 822, 877, 922, 970, 982, 994, and 1006 and remove "and lstate == 3"

I'm still working on many new features so an updated release is significantly delayed, sorry about that. All of these fixes will be in the updated version.

**Comment by OfficerBozza** [comment]
Hey everyone, its been a while since I checked up on this. Keep me updated, looking good so far

**Comment by Internal3247** [comment]
I've been working on some updates and implementing new features, I've completed basically everything and I'll be opening a new PR later tonight when I'm next on my computer with all the features and fixes in this PR/comment thread.

**Comment by Internal3247** [comment]
New PR being made now.

**Pull request closed without merging**